### PR TITLE
fix: warn user about existing deployment during one-shot deploy

### LINKED ISSUE #2873: need to detect that other solo deployments are in an existing cluster prior to running one-shot and halt or warn user
- check for a remote config in another namespace inside the cluster
- add a `--force` to provide just a warning and continue
- without the `--force` then provide information to the user and then exit with an error message

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
## Description

This PR adds a new step to the `solo one-shot single deploy` command. It checks for an existing deployment in the current cluster and prompts the user before proceeding. It also adds a `force` flag to the command, which, if provided, skips the new step altogether.

The PR also adds a new log message group at the end of `solo deployment cluster attach` and `solo one-shot single deploy`, listing all existing deployments in the specified cluster.

### Related Issues

* Closes #2873 

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

**Review by JeffreyDallas — DISMISSED** [review]

File: src/commands/one-shot/default-one-shot.ts [hunk 1]
```diff
@@ -219,12 +221,39 @@ export class DefaultOneShotCommand extends BaseCommand implements OneShotCommand
...
+                const proceed: boolean = await task
+                  .prompt(ListrInquirerPromptAdapter)
+                  .run(confirmPrompt, promptOptions);
+                if (!proceed) {
+                  throw new SoloError('Aborted by user');
+                }
+              }
+            },
+            skip: (context_: OneShotSingleDeployContext): boolean => context_.config.force === true,
+          },
```
[the remark this answers is not in the document]
Comment (Ivo-Yankov) [reply src/commands/one-shot/default-one-shot.ts]: If we only want it to fail in `one-shot`, then I don't see a reason for having the user prompt in `deployment cluster attach`

**Review by jeromy-cannon — APPROVED** [review]
